pkg/api/helper/helpers.go

```diff
@@ -113,9 +113,18 @@ func IsStandardContainerResourceName(str string) bool {
 }
 
 // IsOpaqueIntResourceName returns true if the resource name has the opaque
-// integer resource prefix.
+// integer resource prefix, or it is another unknown resource name not in the
+// default namespace.
 func IsOpaqueIntResourceName(name api.ResourceName) bool {
-	return strings.HasPrefix(string(name), api.ResourceOpaqueIntPrefix)
+	return IsCustomNamespacedResourceName(name)
```

Comment:
Probably not but they will get surprise when they do. What do you think we explicitly check !strings.HasPrefix(string(name), api.ResourceDefaultNamespacePrefix) && !strings.HasPrefix(string(name), "alpha." + api.ResourceDefaultNamespacePrefix) && !strings.HasPrefix(string(name), "beta." + api.ResourceDefaultNamespacePrefix)?